tests/unit_tests/main_test.js

```diff
@@ -40,6 +40,114 @@ context("HintCoordinator", () => {
   });
 });
 
+context("Next zoom level", () => {
+  // NOTE: All these tests use the Chrome zoom levels, which are the default!
+  should("Zoom in 0 times", async () => {
+    const count = 0;
+    const currentZoom = 1.00;
+    const nextZoom = await nextZoomLevel(currentZoom, count);
+    assert.equal(1.00, nextZoom);
+  });
+
+  should("Zoom in 1", async () => {
+    const count = 1;
+    const currentZoom = 1.00;
+    const nextZoom = await nextZoomLevel(currentZoom, count);
+    assert.equal(1.10, nextZoom);
+  });
+
+  should("Zoom out 1", async () => {
+    const count = -1;
+    const currentZoom = 1.00;
+    const nextZoom = await nextZoomLevel(currentZoom, count);
+    assert.equal(0.90, nextZoom);
+  });
+
+  should("Zoom in 2", async () => {
+    const count = 2;
+    const currentZoom = 1.00;
+    const nextZoom = await nextZoomLevel(currentZoom, count);
+    assert.equal(1.25, nextZoom);
+  });
+
+  should("Zoom out 2", async () => {
+    const count = -2;
+    const currentZoom = 1.00;
+    const nextZoom = await nextZoomLevel(currentZoom, count);
+    assert.equal(0.80, nextZoom);
+  });
+
+  should("Zoom in from between values", async () => {
+    const count = 1;
+    const currentZoom = 1.05;
+    const nextZoom = await nextZoomLevel(currentZoom, count);
+    assert.equal(1.10, nextZoom);
+  });
+
+  should("Zoom out from between values", async () => {
+    const count = -1;
+    const currentZoom = 1.05;
+    const nextZoom = await nextZoomLevel(currentZoom, count);
+    assert.equal(1.00, nextZoom);
+  });
+
+  should("Zoom in past the maximum", async () => {
+    const count = 15;
+    const currentZoom = 1.00;
+    const nextZoom = await nextZoomLevel(currentZoom, count);
+    assert.equal(5.00, nextZoom);
+  });
+
+  should("Zoom out past the minimum", async () => {
+    const count = -15;
+    const currentZoom = 1.00;
+    const nextZoom = await nextZoomLevel(currentZoom, count);
+    assert.equal(0.25, nextZoom);
+  });
+
+  should("Zoom in from below the minimum", async () => {
+    const count = 1;
+    const currentZoom = 0.01; // lowest non-broken Chrome zoom level
+    const nextZoom = await nextZoomLevel(currentZoom, count);
+    assert.equal(0.25, nextZoom);
+  });
+
+  should("Zoom out from above the maximum", async () => {
+    const count = -1;
+    const currentZoom = 9.99; // highest non-broken Chrome zoom level
+    const nextZoom = await nextZoomLevel(currentZoom, count);
+    assert.equal(5.00, nextZoom);
+  });
+
+  should("Zoom in from above the maximum", async () => {
+    const count = 1;
+    const currentZoom = 9.99; // highest non-broken Chrome zoom level
```

Comment:
What does non-broken mean? What happens, exactly, if it's > 9.99?

Reply:
It still scales the page correctly to 500%, but it does not parse the number correctly and shows odd % numbers at the top of the page. Note that in the Chrome API documentation for setZoom it does not specify a range of values to be used. I determined the values it handles well from my own testing.
Firefox DOES specify that the value should be between 0.3 and 5, but it gracefully fails for numbers outside of this range. Chrome isn't quite so graceful. That's why I decided to add a limit where Chrome begins to act oddly, but didn't do anything for Firefox's range.

Reply:
Okay, I don't know why I thought it was parsing the number incorrectly.
I think it **does** work correctly for bigger values, it's just that there isn't enough space to show the number, so it shows "1,0..." because the "1,000" gets truncated. And it looks like for smaller values it just does nothing, similar to Firefox.
I must have been thinking incorrectly before.
So this bounds checking may be unnecessary. I am going to remove it.